Move roles to CONTRIBUTING.rst (#10327)
mistercrunch/Airflow: CONTRIBUTING.rst

.. contents:: :local:

Roles
=============

There are several roles within the Airflow Open-Source community.

PMC Member
-----------
The PMC (Project Management Committee) is a group of maintainers that drives changes in the way that
Airflow is managed as a project.

Considering Apache, the role of the PMC is primarily to ensure that Airflow conforms to Apache's processes
and guidelines.

Committers/Maintainers
----------------------

Committers are community members that have write access to the project’s repositories, i.e., they can modify the code,
documentation, and website by themselves and also accept other contributions.

The official list of committers can be found `here <https://airflow.apache.org/docs/>`__.

Additionally, committers are listed in a few other places (some of these may only be visible to existing committers):

* https://whimsy.apache.org/roster/ppmc/airflow
* https://github.com/orgs/apache/teams/airflow-committers/members

Committers are responsible for:

* Championing one or more items on the `Roadmap <https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Home>`__
* Reviewing & Merging Pull-Requests

> * Scanning and responding to JIRAs

Reviewer: ```suggestion
* Scanning and responding to Github issues
```

Reviewer 2: +1

Author: Done.

> * Responding to questions on the dev mailing list (dev@airflow.incubator.apache.org)

Reviewer: ```suggestion
* Responding to questions on the dev mailing list (dev@airflow.apache.org)
```

Reviewer 2: +1

Author: done.

Becoming a Committer
--------------------

There is no strict protocol for becoming a committer.
Candidates for new committers are typically people that are active contributors and community members.

The key aspects of a committer are:

* Consistent contributions over the past 6 months

> * Expressed understanding of Airflow Core or has displayed holistic understanding of a particular part and made

Reviewer: ```suggestion
* Understanding of Airflow Core or has displayed a holistic understanding of a particular part and made
```

Reviewer 2: yep

Author: done.

  contributions towards a more strategic goal

> * Understands contributor/committer guidelines: `Contributors' Guide <https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst>`__

Reviewer: ```suggestion
* Understanding of contributor/committer guidelines: `Contributors' Guide <https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst>`__
```

Reviewer 2: yep

Author: done.

> * Quality of the commits, and asses the impact of the changes

Reviewer: What does it mean `asses the impact of the changes`?

Author: As @potiuk suggested, I removed that part.

> * Quality of the commits, and asses the impact of the changes

Reviewer 2: ```suggestion
* Quality of the commits

Author: Removed.

> * Visibility on discussions on the dev mailing list and the Slack channel

Reviewer: ```suggestion
* Visibility in community discussions (on the dev mailing list, Slack and Github)
```
WDYT?

Reviewer 2: +1

Author: @turbaszek Agreed.

> * Testing Release Candidates to improve the release cycle

Reviewer: ```suggestion
* Testing Release Candidates
```

Reviewer 2: +1

Author: done.

-   Remember that this is a volunteer-driven project, and that contributions are
    welcome :)

Roles
=============

There are several roles within the Airflow Open-Source community.

PMC Member
-----------
The PMC (Project Management Committee) is a group of maintainers that drives changes in the way that
Airflow is managed as a project.

Considering Apache, the role of the PMC is primarily to ensure that Airflow conforms to Apache's processes
and guidelines.

Committers/Maintainers
----------------------

Committers are community members that have write access to the project’s repositories, i.e., they can modify the code,
documentation, and website by themselves and also accept other contributions.

> The official list of committers can be found `here <https://airflow.apache.org/docs/>`__.

Reviewer 3: ```suggestion
The official list of committers can be found `here <https://airflow.apache.org/docs/stable/project.html#committers>`__.
```

.. code-block:: bash

>    ./breeze initialize-local-virtualenv

Reviewer: Is this a related change?

Reviewer 2: I think not related but I would not obiect fixing it.

Author: Not related, just a typo I noticed when trying to use Breeze for the first time. If no one opposes, I'll keep it here.
